### ISSUE TITLE
http_build_query 

### ISSUE THREAD

**Issue by elvispdosreis** [guzzle/guzzle#3126]
**Guzzle version(s) affected**: 6.5.8
**PHP version**: 8.1

**Description**  
http_build_query(): Passing null to parameter #2 ($numeric_prefix) of type string is deprecated
guzzlehttp/guzzle/src/Client.php line 445

**How to reproduce** 
``` 
if (is_array($value)) {
 $value = http_build_query($value, null, '&', PHP_QUERY_RFC3986);
}
```

**Possible Solution**  
```
if (is_array($value)) {
 $value = http_build_query($value, '', '&', PHP_QUERY_RFC3986);
}
```

**Comment by mfn** [comment]
See https://github.com/guzzle/guzzle#version-guidance , 6.x is in security-fixes mode and only supports PHP <8.0

**Comment by GrahamCampbell** [comment]
Yep, that's correct. Please upgrade to the latest version of Guzzle 7 ASAP.
